test/systemtests/aci_test.go

```diff
@@ -72,403 +73,582 @@ func (s *systemtestSuite) TestACIMode(c *C) {
+		c.Assert(s.checkACILearning(s.globInfo.Tenant,
+			"profile1",
+			"epga",
+			containersA), IsNil)
 
-	s.baremetal.IterateNodes(func(node vagrantssh.TestbedNode) error {
-		node.RunCommand("sudo rm /tmp/*net*")
-		return node.RunCommand("docker pull alpine")
-	})
+		c.Assert(s.checkACILearning(s.globInfo.Tenant,
+			"profile1",
+			"epgb",
+			containersB), IsNil)
+
+		c.Assert(s.startListeners(containersA, []int{8000, 8001}), IsNil)
+		c.Assert(s.startListeners(containersB, []int{8000, 8001}), IsNil)
+		c.Assert(s.checkConnectionPairRetry(containersA, containersB, 8000, 1, 3), IsNil)
+		c.Assert(s.checkNoConnectionPairRetry(containersA, containersB, 8001, 1, 3), IsNil)
+		for _, cB := range containersB {
+			c.Assert(s.pingTestByName(containersA, cB.eth0.ip), IsNil)
+		}
+
+		log.Infof("Restarting ACI gateway on all nodes")
+		for iNode := range s.nodes {
+			s.nodes[iNode].tbnode.RunCommand("sudo service aci-gw restart")
+		}
+		time.Sleep(10 * time.Millisecond)
+
+		//Verifying the behavior didn't change after restarting the gateway
+		for _, cB := range containersB {
+			c.Assert(s.pingTestByName(containersA, cB.eth0.ip), IsNil)
+		}
+		c.Assert(s.pingTestByName(containersA, s.globInfo.Gateway), IsNil)
+		c.Assert(s.pingTestByName(containersB, s.globInfo.Gateway), IsNil)
+		c.Assert(s.pingTest(containersA), IsNil)
+		c.Assert(s.pingTest(containersB), IsNil)
 
-	//Copying binaries
-	s.copyBinary("netmaster")
-	s.copyBinary("netplugin")
-	s.copyBinary("netctl")
-	s.copyBinary("contivk8s")
+		// Add a rule to allow 8001
+		c.Assert(s.cli.RulePost(&client.Rule{
+			RuleID:            "3",
+			PolicyName:        "policyAB",
+			TenantName:        s.globInfo.Tenant,
+			FromEndpointGroup: "epga",
+			Direction:         "in",
+			Protocol:          "tcp",
+			Port:              8001,
+			Action:            "allow",
+		}), IsNil)
+		time.Sleep(time.Second * 10)
+
+		c.Assert(s.checkACILearning(s.globInfo.Tenant,
+			"profile1",
+			"epga",
+			containersA), IsNil)
+
+		c.Assert(s.checkACILearning(s.globInfo.Tenant,
+			"profile1",
+			"epgb",
+			containersB), IsNil)
 
-}*/
+		c.Assert(s.checkConnectionPairRetry(containersA, containersB, 8000, 1, 3), IsNil)
+		c.Assert(s.checkConnectionPairRetry(containersA, containersB, 8001, 1, 3), IsNil)
+
+		// Delete the app profile
+		c.Assert(s.cli.AppProfileDelete(s.globInfo.Tenant, "profile1"), IsNil)
+		time.Sleep(time.Second * 5)
+
+		c.Assert(s.removeContainers(append(containersA, containersB...)), IsNil)
+		containersA = nil
+		containersB = nil
+		c.Assert(s.cli.EndpointGroupDelete(s.globInfo.Tenant, "epga"), IsNil)
+		c.Assert(s.cli.EndpointGroupDelete(s.globInfo.Tenant, "epgb"), IsNil)
+		c.Assert(s.cli.RuleDelete(s.globInfo.Tenant, "policyAB", "2"), IsNil)
+		c.Assert(s.cli.RuleDelete(s.globInfo.Tenant, "policyAB", "3"), IsNil)
+
+		c.Assert(s.cli.PolicyDelete(s.globInfo.Tenant, "policyAB"), IsNil)
+		c.Assert(s.cli.NetworkDelete(s.globInfo.Tenant, s.globInfo.Network), IsNil)
+	}
+}
```

Comment:
Ideally all the ACI tests should conform to the existing tests. 
If we have taken the config file approach. we need to mention the physical networks information. Which should be used to iterate and derive the number of epgs,networks policies.  This can be part of a seperate PR. file an issue for this after PR is merged